mcs/class/referencesource/mscorlib/system/globalization/charunicodeinfo.cs

```diff
@@ -188,14 +188,12 @@ unsafe private static double EndianSwap(double value)
             // Go to native side and get pointer to the native table
             byte * pDataTable = GlobalizationAssembly.GetGlobalizationResourceBytePtr(typeof(CharUnicodeInfo).Assembly, UNICODE_INFO_FILE_NAME);
 
-            UnicodeDataHeader* mainHeader = (UnicodeDataHeader*)pDataTable;
-
             // Set up the native pointer to different part of the tables.
-            s_pCategoryLevel1Index = (ushort*) (pDataTable + EndianSwap(mainHeader->OffsetToCategoriesIndex));
```

Comment:
This is reference source. Maybe you should check for `#if MONO` or `#if UNITY`. Please as in Slack chat with Xamarin